python/task_manager/background_task_manager.py

```diff
@@ -325,15 +429,13 @@ def _get_worker_thread(self):
         # create a new worker thread - note, there are two different implementations of the WorkerThread class
         # that use two different recipes.  Although WorkerThreadB is arguably more correct it has some issues
         # in PyQt so WorkerThread is currently preferred - see the notes in the class above for further details
-        thread = WorkerThread(self)
+        thread = WorkerThread(self._results_poller)
```

Comment:
i see, the constructor changed to `def __init__(self, rp, parent=None)` !

I think we should pass the parent too - i suspect there may be issues otherwise:
```
thread = WorkerThread(self._results_poller, parent=self)
```